### PR TITLE
Update docs to reference new interface

### PR DESCRIPTION
**Pull request by BenjaminDev**
## Description
- Updated the docs to reference the new python interface where needed.
- Updated sphinx to use 3.2.1 and 3.2.0 was broken and we pinned below that. Now 3.2.1 is out we can safely update it. 

## Affected Dependencies
Sphinx (just for the docs not a `python-dp` dependency) 
## How has this been tested?
- All docs build and render locally when following the instructions in `contributing.md`
## Checklist
- [x] I have followed the [Contribution Guidelines](https://github.com/OpenMined/.github/blob/master/CONTRIBUTING.md) and [Code of Conduct](https://github.com/OpenMined/.github/blob/master/CODE_OF_CONDUCT.md)
- [x] I have commented my code following the [OpenMined Styleguide](https://github.com/OpenMined/.github/blob/master/STYLEGUIDE.md)
- [ ] I have labeled this PR with the relevant [Type labels](https://github.com/OpenMined/.github/labels?q=Type%3A)
- [ ] My changes are covered by tests

### CODE REVIEW

File: docs/requirements.txt [hunk 1]
```diff
@@ -1,3 +1,3 @@
 python-dp
 sphinx-rtd-theme
-sphinx<=3.1.2 # because of https://github.com/sphinx-doc/sphinx/issues/8074
\ No newline at end of file
+sphinx>=3.2.1
```
Comment (chinmayshah99) [inline docs/requirements.txt:3]: Is this issue not relevant to us?

Comment (chinmayshah99) [reply docs/requirements.txt:3]: https://github.com/OpenMined/PyDP/pull/238

Comment (BenjaminDev) [reply docs/requirements.txt:3]: Nope: `broken sphinx 3.2.0 has been fixed in sphinx 3.2.1. `
We can safely upgrade.

**Review by chinmayshah99 — APPROVED** [review]
